staging/src/k8s.io/apiserver/pkg/registry/rest/rest.go

```diff
@@ -207,6 +207,26 @@ type UpdatedObjectInfo interface {
 	UpdatedObject(ctx context.Context, oldObj runtime.Object) (newObj runtime.Object, err error)
 }
 
+// UpdateConfig provides information about the validation functions that
+// the Updater should use, and allows for overriding certain UpdateStrategy behavior.
+type UpdateConfig interface {
```

Comment:
why do we need this abstraction? Why not a struct?

Reply:
It enforces that the constructor is called on the private struct. But other than that it feels overabstracted.

Reply:
I think that having it as an interface is the only way to make it possible to use the builder pattern,
suggested by [this comment](https://github.com/kubernetes/kubernetes/pull/65154#discussion_r195877249)
> What do you think about not taking "forceAllowCreateOnUpdate" as a parameter and instead having a .SetForceAllowCreateOnUpdate() method (builder pattern) to reduce call site churn when we add uses of this?